src/test/java/org/springframework/data/elasticsearch/repository/query/ElasticsearchStringQueryUnitTests.java

```diff
@@ -138,6 +141,27 @@ private org.springframework.data.elasticsearch.core.query.Query createQuery(Stri
 		return elasticsearchStringQuery.createQuery(new ElasticsearchParametersParameterAccessor(queryMethod, args));
 	}
 
+	@Test // #2062
+	@DisplayName("create source filter for query")
+	void shouldCreateSourceFilter() {
+		ElasticsearchQueryMethod queryMethod = mock(ElasticsearchQueryMethod.class);
+		when(queryMethod.hasIncludes()).thenReturn(true);
+		when(queryMethod.getIncludes()).thenReturn("?0");
+		when(queryMethod.getExcludes()).thenReturn("?1");
+
+		ElasticsearchStringQuery elasticsearchStringQuery = spy(new ElasticsearchStringQuery(
+				queryMethod,
+				operations,
+				"{\"match_all\": {}}"
+		));
+		ParametersParameterAccessor parameterAccessor = mock(ParametersParameterAccessor.class);
+		when(parameterAccessor.getBindableValue(0)).thenReturn("foo");
+		when(parameterAccessor.getBindableValue(1)).thenReturn("bar");
+		SourceFilter resp = elasticsearchStringQuery.createSourceFilter(parameterAccessor);
+		assertThat(resp.getIncludes()).isEqualTo(new String[]{"foo"});
+		assertThat(resp.getExcludes()).isEqualTo(new String[]{"bar"});
+	}
+
```

Comment:
This is testing the creation of the `SourceFilter`.

What is missing is testing if the values are passed from the `@Query` annotation to the `ElasticsearchQueryMethod` and that the created Query in the end really has the filters set.

I think it would make sense for this use case to have an integration test where it is checked that the values configured in the `@Query` annotation in combination with some method parameters do have the desired response from the ES cluster.

Reply:
yes I agree, I will do the intergration tests soon